cpp/daal/src/algorithms/svm/svm_predict_impl.i

```diff
@@ -191,30 +200,58 @@ private:
 template <typename algorithmFPType, CpuType cpu>
 struct SVMPredictImpl<defaultDense, algorithmFPType, cpu> : public Kernel
 {
-    services::Status compute(const NumericTablePtr & xTable, Model * model, NumericTable & r, const svm::Parameter * par)
-    {
-        kernel_function::KernelIfacePtr kernel = par->kernel->clone();
-        DAAL_CHECK(kernel, ErrorNullParameterNotSupported);
-
-        const NumericTablePtr svCoeffTable = model->getClassificationCoefficients();
-        const NumericTablePtr svTable      = model->getSupportVectors();
-
-        const algorithmFPType bias(model->getBias());
-
-        const size_t nVectors = xTable->getNumberOfRows();
-        const size_t nSV      = svTable->getNumberOfRows();
+    using TPredictTask = PredictTask<algorithmFPType, cpu>;
 
-        const size_t nRowsPerBlock = 128;
-        const size_t nBlocks       = nVectors / nRowsPerBlock + !!(nVectors % nRowsPerBlock);
-
-        size_t nSVPerBlock = 0;
-        DAAL_SAFE_CPU_CALL((nSVPerBlock = 128), (nSVPerBlock = nSV));
-        const size_t nBlocksSV = nSV / nSVPerBlock + !!(nSV % nSVPerBlock);
-
-        const bool isSparse = xTable->getDataLayout() == NumericTableIface::csrArray;
+    services::Status computeSequential(const NumericTablePtr & xTable, const NumericTablePtr & svCoeffTable, const NumericTablePtr & svTable,
+                                       NumericTable & r, kernel_function::KernelIfacePtr & kernel, const algorithmFPType bias, const size_t nVectors,
+                                       const size_t nSV, const bool isSparse)
+    {
+        services::Status st;
+        TArray<algorithmFPType, cpu> distances(nVectors);
```

Comment:
Please place the memory allocation check for `distances` array right after its definition.

Reply:
TArray declaration was moved closer to existing check